src/storage.rs

```diff
@@ -121,6 +123,48 @@ impl CQLSession {
         self.0.query(CREATE_EDGE_TABLE_QUERY)?;
         Ok(())
     }
+
+    fn paged_query(
+        &self,
+        query: String,
+        values: QueryValues,
+        page_size: i32,
+    ) -> Result<Vec<Row>, CDRSError> {
+        let mut rows: Vec<Row> = Vec::new();
+        let mut pager_state: PagerState = PagerState::new();
+        loop {
+            let mut params = QueryParamsBuilder::new().page_size(page_size);
+            if pager_state.get_cursor().is_some() {
+                params = params.paging_state(pager_state.get_cursor().clone().unwrap());
+            }
+            params = params.values(values.clone());
+
+            let body = self
+                .0
+                .query_with_params(query.clone(), params.finalize())
+                .and_then(|frame| frame.get_body())?;
+
+            let metadata_res: error::Result<RowsMetadata> = body
+                .as_rows_metadata()
+                .ok_or("Pager query should yield a vector of rows".into());
+            let metadata = metadata_res?;
+
+            let has_more_pages =
+                Some(RowsMetadataFlag::has_has_more_pages(metadata.flags.clone())).unwrap();
```

Comment:
This unwrap is not needed. Please use other methods or `if let` to match.

Reply:
Modified. Thank you for pointing this out.